service/sharddistributor/config/dynamic_config.go

```diff
@@ -0,0 +1,42 @@
+package config
+
+import (
+	"github.com/uber/cadence/common/dynamicconfig"
+	"github.com/uber/cadence/common/dynamicconfig/dynamicproperties"
+	"github.com/uber/cadence/common/types"
+)
+
+// DynamicConfig represents dynamic configuration for shard distributor service
+type DynamicConfig struct {
+	LoadBalancingMode dynamicproperties.StringPropertyFnWithNamespaceFilters
+	MigrationMode     dynamicproperties.StringPropertyFnWithNamespaceFilters
+}
```

Comment:
The rest of cadence parses the dynamic config properties raw, I dont think we should introduce a new concept.